src/Castle.ts

```diff
@@ -0,0 +1,218 @@
+import fetch from 'node-fetch';
+import { reduce } from 'lodash';
+import { IncomingHttpHeaders } from 'http2';
+import AbortController from 'abort-controller';
+
+const defaultApiUrl = 'https://api.castle.io';
+
+type TrackParameters = {
```

Comment:
I would call it EventParameters or differently because they are not used only for tracking

Reply:
EventParameters doesn't make sense, since event is a part of the parameters. Maybe just `Parameters` or like `ActionParameters`?